lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2468,6 +2468,49 @@ static LogicalResult verifyHWStructCastOp(HWStructCastOp cast) {
   return success();
 }
 
+static LogicalResult verifyBitCastOp(BitCastOp cast) {
+  StringRef errorMsg;
+  // Infer the bitwidth of the type, if it can be computed. Else return false.
+  std::function<bool(FIRRTLType, size_t &)> getWidth =
+      [&](FIRRTLType type, size_t &width) -> bool {
+    return TypeSwitch<FIRRTLType, size_t>(type)
+        .Case<BundleType>([&](auto bundle) {
+          for (auto &elt : bundle.getElements())
+            if (!getWidth(elt.type, width))
+              return false;
+          return true;
+        })
+        .Case<FVectorType>([&](auto vector) {
+          for (size_t i = 0, e = vector.getNumElements(); i != e; ++i)
+            if (!getWidth(vector.getElementType(), width))
+              return false;
+          return true;
+        })
+        .Case<IntType>([&](IntType iType) {
+          if (!iType.getWidth().hasValue()) {
+            errorMsg = "valid bitwidth is unknown";
+            return false;
+          }
+          width += iType.getWidth().getValue();
+          return true;
+        })
+        .Default([&](auto t) {
+          errorMsg = "bitcastOp only supports types with valid bit width:";
```

Comment:
Why doesn't it handle clock and other things?  Should we use this instead of the existing "as clock" casts?

Reply:
Right, we already have `AsSIntPrimOp`, `AsUIntPrimOp`, `AsClockPrimOp`, `AsAsyncResetPrimOp` to handle the other cases.